### PR TITLE
added data types

### PR DESCRIPTION
**Pull request by TeiturMcSwagger**
Added data types

---
This change is [Reviewable](https://reviewable.io/reviews/ctxd/freetacos/2)

### CODE REVIEW

File: src/error_id_action.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        IGNORE      = 0;
+        SHUTDOWN    = 1;
+        RESET       = 2;
+    }
+
+class ErrorIDAction
+{
+    private:
+        int ErrorIdentifier;                    /* required */
+        std::optional <NameType> Description;   /* optional */
```
Comment (ruttam) [inline src/error_id_action.h]: Consider deleting a space ;)

File: src/partition.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include <general_types.h>
```
Comment (ruttam) [inline src/partition.h]: Consider structuring includes in a readable (short to long imports) order

File: src/process.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include <general_types.h>
+
+class Process
+{
+    private:
+        char[MAXLENGTH] name;                           /* required */
```
Comment (ruttam) [inline src/process.h]: Is it a NameType?

**Review by ruttam — CHANGES_REQUESTED** [review]

**Review by TeiturMcSwagger — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ctxd/freetacos/2)* status: 0 of 10 files reviewed, 3 unresolved discussions (waiting on @ruttam)

___
*[src/error_id_action.h, line 15 at r1](https://reviewable.io/reviews/ctxd/freetacos/2#-LqGmpFoX_KiFD_WXXNL-r1-15:-LqGsnSl7xFYukdkX-R4:b-896fix) ([raw file](https://github.com/ctxd/freetacos/blob/e0496cb79c76983266cb7c95faf4b0f64cac7327/src/error_id_action.h#L15)):*
<details><summary><i>Previously, ruttam (Rūta) wrote…</i></summary><blockquote>

Consider deleting a space ;)
</blockquote></details>

Done.
___
*[src/partition.h, line 1 at r1](https://reviewable.io/reviews/ctxd/freetacos/2#-LqGmpFoX_KiFD_WXXNO-r1-1:-LqGsozN1xYj7d8X078K:b-896fix) ([raw file](https://github.com/ctxd/freetacos/blob/e0496cb79c76983266cb7c95faf4b0f64cac7327/src/partition.h#L1)):*
<details><summary><i>Previously, ruttam (Rūta) wrote…</i></summary><blockquote>

Consider structuring includes in a readable (short to long imports) order
</blockquote></details>

Done.
___
*[src/process.h, line 6 at r1](https://reviewable.io/reviews/ctxd/freetacos/2#-LqGmpFoX_KiFD_WXXNQ-r1-6:-LqGsqPbD2rk4MS4XG9d:b-896fix) ([raw file](https://github.com/ctxd/freetacos/blob/e0496cb79c76983266cb7c95faf4b0f64cac7327/src/process.h#L6)):*
<details><summary><i>Previously, ruttam (Rūta) wrote…</i></summary><blockquote>

Is it a NameType?
</blockquote></details>

Done.

**Pull request merged**